danceradvent/public/articles/2023/20-dancing-with-htmx.pod

```diff
@@ -0,0 +1,333 @@
+the third displays some icons that allow the user to interact with the record
+in various ways. The first allows you to edit the name, the second will (when
+the code is implemented) display a list of the presents the child will receive
+and the third deletes the child from the database. Let's start by looking at
+the easiest option, the delete icon.
+
+There are three C<hx-*> attributes on the delete icon:
+
+=over 4
+
+=item hx-delete
+
+This defines the action that clicking on this icon will take. There are
+C<hx-VERB> attributes for all of the HTTP verbs - C<GET>, C<POST>, C<PUT>,
+C<PATCH> and C<DELETE>. In this case we're using C<hx-delete> because that's
+the correct verb to use to delete a resource. When the icon is clicked, HTMX
+will generate an HTTP C<DELETE> request to C</delete/ID> (where ID is the
+ID of the database row). This means that we'll need to write a route in our
+Dancer app that matches C<del '/delete/:id'>. We'll see the code for that
+in a minute.
+
+=item hx-confirm
+
+This is a nice feature that HTMX gives us. Adding an C<hx-confirm> attribute
+to our HTML tells HTMX to intercept a click on the icon and display a pop-up
+message in the browser. The pop-up will have OK and Cancel buttons and the 
+action will only proceed if the OK button is selected. This is a simple way
+to add a confirmation step to potentially destructive actions.
+
+=item hx-target
+
+When HTMX makes a request to the server, it expects to receive HTML back. By
+default, it will replace the contents of the element that triggered the
+request (by, for example, being clicked) with the HTML that was returned. In
+my experience, this is rarely what you want so you can change the element
+that is replaced by defining it using a CSS selector in an C<hx-target>
+attribute. Usually, the simplest way to do this is to put an ID on the element
+you want to replace.
+
+In this case, having deleted a row from the database, we want to replace the
+element containing the list of rows (because it will be one item shorter).
+So we give it a target of C<#list-body> and ensure that ID is defined on the
+correct HTML element (as you'll see in the C<list.tt> template above).
+
+=back
+
+So, we're in the situation where clicking the delete icon will take the
+following steps:
+
+=over 4
+
+=item *
+
+Present a confirmation dialogue.
+
+=item *
+
+Make an HTTP C<DELETE> request to C</delete/ID>.
+
+=item *
+
+Expect to get an HTML fragment back from the server and replace the
+C<#list_body> element with that HTML.
+
+=back
+
+All we need to do now is to write the correct route in our Dancer app. The
+code looks like this:
+
+    del '/delete/:id' => sub {
+      my $id = route_parameters->get('id');
+
+      return 404 unless $id;
+      return 404 if $id =~ /\D/;
+
+      my $sch = HTMXmas::Schema->connect(...);
+      my $rs  = $sch->resultset('Child');
+
+      $rs->find($id)->delete;
+
+      $row->delete;
```

Comment:
Did you originally have one line to find and another to delete? I think you can safely delete this.

Reply:
I went back and forth on those lines so many times. And ended up in an inconsistent state :-0